Framework/DataHandling/src/GroupDetectors2.cpp

```diff
@@ -938,40 +938,21 @@ size_t GroupDetectors2::formGroups(API::MatrixWorkspace_const_sptr inputWS,
     // are assumed to be the same here
     outSpec.setSharedX(inputWS->sharedX(0));
 
-    // the Y values and errors from spectra being grouped are combined in the
-    // output spectrum
-    auto &firstY = outSpec.mutableY();
     // Keep track of number of detectors required for masking
     size_t nonMaskedSpectra(0);
-    beh->mutableX(outIndex)[0] = 0.0;
-    beh->mutableE(outIndex)[0] = 0.0;
+
     for (auto originalWI : it->second) {
       // detectors to add to firstSpecNum
-      const auto &fromSpectrum = inputWS->getSpectrum(originalWI);
+      const auto &inputSpectrum = inputWS->getSpectrum(originalWI);
 
-      // Add up all the Y spectra and store the result in the first one
+      outSpec.setHistogram(outSpec.histogram() += inputSpectrum.histogram());
+      outSpec.addDetectorIDs(inputSpectrum.getDetectorIDs());
 
-      firstY += fromSpectrum.y();
-
-      auto fEit = outSpec.mutableE().begin();
-      auto Eit = fromSpectrum.e().cbegin();
-      for (auto fYit = firstY.begin(); fYit != firstY.end();
-           ++fYit, ++fEit, ++Eit) {
-        // Assume 'normal' (i.e. Gaussian) combination of errors
-        *fEit = std::sqrt((*fEit) * (*fEit) + (*Eit) * (*Eit));
-      }
-
-      // detectors to add to the output spectrum
-      outSpec.addDetectorIDs(fromSpectrum.getDetectorIDs());
-      try {
-        Geometry::IDetector_const_sptr det = inputWS->getDetector(originalWI);
-        if (!det->isMasked())
-          ++nonMaskedSpectra;
-      } catch (Exception::NotFoundError &) {
-        // If a detector cannot be found, it cannot be masked
+      if (!isMaskedDetector(*inputWS->getDetector(originalWI))) {
```

Comment:
Does `SpectrumInfo` throw as I might just change the call in the wrapper if it still does?